crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2016,12 +2016,50 @@ impl<'db> Type<'db> {
             | Type::FunctionLiteral(..) => {
                 let fallback = self.instance_member(db, name_str);
 
-                self.invoke_descriptor_protocol(
+                let result = self.invoke_descriptor_protocol(
                     db,
                     name_str,
                     fallback,
                     InstanceFallbackShadowsNonDataDescriptor::No,
-                )
+                );
+
+                let custom_getattr_result =
+                    || {
+                        // Typeshed has a fake `__getattr__` on `types.ModuleType` to help out with dynamic imports.
+                        // We explicitly hide it here to prevent arbitrary attributes from being available on modules.
+                        if self.into_instance().is_some_and(|instance| {
+                            instance.class.is_known(db, KnownClass::ModuleType)
+                        }) {
+                            return Symbol::Unbound.into();
+                        }
```

Comment:
We do this elsewhere, so I took the same approach here. I'm not 100% if this is justified here, though.

Reply:
Yes, I think this makes sense here. We use `ModuleType` to model attribute access fallbacks for known modules ("module literals"), and the typeshed definition of `ModuleType` is more geared towards "arbitrary unknown module object."